src/diagnostics.rs

```diff
@@ -185,6 +195,64 @@ fn absolute_path(path_str: &str) -> PathBuf {
     current_dir().unwrap().join(path).clean()
 }
 
+/// syntax checking using verilator --lint-only
+fn verilator_syntax(
+    rope: &Rope,
+    verilator_syntax_uri: &Url,
+    verilator_syntax_path: &str,
+    verilator_syntax_args: &[String],
+) -> Option<Vec<Diagnostic>> {
+    let mut child = Command::new(verilator_syntax_path)
+        .stdin(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .args(verilator_syntax_args)
+        .arg(verilator_syntax_uri.to_file_path().ok()?.to_str()?)
+        .spawn()
+        .ok()?;
+    let re =
```

Comment:
I set a bad example but it doesn't seem good to compile this regex, every time we get diagnostics.
It's possible that this overhead is significantly less than actually running verilator so I'll leave it up to you.